### PR TITLE
treewide: rename name to pname&version

### PR DESCRIPTION
**Pull request by Stunkymonkey**
###### Motivation for this change
#103997

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.05 Release Notes (or backporting 21.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2205-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/libraries/vte/2.90.nix [hunk 1]
```diff
@@ -1,14 +1,11 @@
 { lib, stdenv, fetchurl, intltool, pkg-config, glib, gtk3, ncurses, gobject-introspection }:
 
 stdenv.mkDerivation rec {
-  versionMajor = "0.36";
-  versionMinor = "3";
-  moduleName   = "vte";
-
-  name = "${moduleName}-${versionMajor}.${versionMinor}";
+  version = "0.36.3";
+  pname = "vte";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/vte/2.90.nix:5]:
```suggestion
  pname = "vte";
  version = "0.36.3";
```

File: pkgs/development/libraries/tclx/default.nix [hunk 2]
```diff
@@ -1,18 +1,17 @@
...
   src = fetchurl {
-    url = "mirror://sourceforge/tclx/tclx${version}.${patch}.tar.bz2";
+    url = "mirror://sourceforge/tclx/tclx${version}.tar.bz2";
     sha256 = "1v2qwzzidz0is58fd1p7wfdbscxm3ip2wlbqkj5jdhf6drh1zd59";
   };
 
   # required in order for tclx to properly detect tclx.tcl at runtime
   postInstall = ''
-    ln -s $prefix/lib/tclx${version} $prefix/lib/tclx${version}/tclx${version}
+    ln -s $prefix/lib/tclx${lib.versions.majorMinor version} $prefix/lib/tclx${lib.versions.majorMinor version}/tclx${lib.versions.majorMinor version}
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/tclx/default.nix:14]: Lets move this into a let in to also keep the line a bit shorter.

File: pkgs/development/libraries/science/math/clmagma/default.nix [hunk 3]
```diff
@@ -32,8 +31,9 @@ let
     INC       = -I$(clBLAS)/include
                #-I$(AMDAPP)/include
   '';
-in stdenv.mkDerivation {
-  name = "clmagma-${version}";
+in stdenv.mkDerivation rec {
+  pname = "clmagma";
+  version = "1.3.0";
   src = fetchurl {
     url = "http://icl.cs.utk.edu/projectsfiles/magma/cl/clmagma-${version}.tar.gz";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/science/math/clmagma/default.nix:38]:
```suggestion
    url = "https://icl.cs.utk.edu/projectsfiles/magma/cl/clmagma-${version}.tar.gz";
```
Unrelated but the site supports https.

File: pkgs/development/libraries/qt-4.x/4.8/default.nix [hunk 4]
```diff
@@ -18,19 +18,19 @@
...
 in
 
 # TODO:
 #  * move some plugins (e.g., SQL plugins) to dedicated derivations to avoid
 #    false build-time dependencies
 
 stdenv.mkDerivation rec {
-  name = "qt-${vers}";
+  pname = "qt";
+  version = "${v_maj}.${v_min}";
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/qt-4.x/4.8/default.nix:29]: Could we also remove v_maj and v_min?

File: pkgs/development/libraries/prison/default.nix [hunk 5]
```diff
@@ -1,12 +1,12 @@
 { lib, stdenv, fetchurl, cmake, qrencode, qt4, libdmtx }:
 
-let v = "1.0"; in
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/prison/default.nix:3]:
```suggestion
```

File: pkgs/development/libraries/physics/geant4/datasets.nix [hunk 6]
```diff
@@ -1,21 +1,21 @@
 { lib, stdenv, fetchurl, geant_version }:
 
 let
-  mkDataset = { name, version, sha256, envvar }:
+  mkDataset = { pname, version, sha256, envvar }:
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/physics/geant4/datasets.nix:4]: Is there an update script we need to adjust?

File: pkgs/development/libraries/libsnark/default.nix [hunk 7]
```diff
@@ -1,18 +1,17 @@
 { lib, stdenv, fetchFromGitHub, cmake, pkg-config, openssl, boost, gmp, procps }:
 
 let
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libsnark/default.nix:3]:
```suggestion
```

File: pkgs/development/libraries/libsnark/default.nix [hunk 8]
```diff
@@ -1,18 +1,17 @@
...
-  rev = "9e6b19ff15bc19fba5da1707ba18e7f160e5ed07";
 in stdenv.mkDerivation rec {
```
Comment (SuperSandro2000) [inline pkgs/development/libraries/libsnark/default.nix:4]:
```suggestion
stdenv.mkDerivation rec {
```

File: pkgs/development/libraries/physics/geant4/datasets.nix (hunk 6 above)
Comment (Stunkymonkey) [reply pkgs/development/libraries/physics/geant4/datasets.nix:4]: i am unable to find one.

**Pull request merged**